lib/remote/configstageshandler.hpp

```diff
@@ -47,6 +47,7 @@ class ConfigStagesHandler final : public HttpHandler
 		const Dictionary::Ptr& params
 	);
 
+	static bool m_RunningPackageUpdates;
```

Comment:
Is this variable guarded by any lock? (Or better directly use a mutex. Hint: have a look at `std::unique_lock` and `std::try_to_lock`)

Reply:
> Or better directly use a mutex.

"directly"?

Reply:
Use it as if it was a `std::atomic<bool>` that can conveniently be used with `std::unique_lock` (but not an option anyways if unlocking from another thread is required).